From 9c38d441c41bbffbc06af09b0d821171eeb43553 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 17 Jan 2023 18:04:30 +0100
Subject: [PATCH] zlib: Remove unsetting _FILE_OFFSET_BITS

Apply https://github.com/madler/zlib/pull/764 on bundled zlib to fix
build with time64.inc as in:
https://git.openembedded.org/openembedded-core/tree/meta/conf/distro/include/time64.inc?id=18df5694796bcfee73c3765bc991bcef055466e3

FAILED: obj/third_party/zlib/zlib/gzclose.o
arm-webos-linux-gnueabi-gcc  -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a8 -D_TIME_BITS=64 -D_FILE_OFFSET_BITS=64 --sysroot=/OE/build/luneos-mickledore/webos-ports/tmp-glibc/work/cortexa8t2hf-neon-webos-linux-gnueabi/webruntime/94-r59.2/recipe-sysroot -MMD -MF obj/third_party/zlib/zlib/gzclose.o.d -DENABLE_BROWSER_CONTROL_WEBAPI=1 -DENABLE_SAMPLE_WEBAPI=1 -DENABLE_MEMORYMANAGER_WEBAPI=1 -DENABLE_WEBOS_SERVICE_BRIDGE_WEBAPI=1 -DENABLE_WEBOS_SYSTEM_WEBAPI=1 -DNEVA_DCHECK_ALWAYS_ON=1 -DOZONE_PLATFORM_WAYLAND_EXTERNAL=1 -DNEVA_OZONE_PLATFORM_WAYLAND=1 -DOS_WEBOS=1 -DNEVA_VIDEO_HOLE=1 -DUSE_NEVA_APPRUNTIME=1 -DUSE_NEVA_BROWSER_SERVICE=1 -DUSE_WEBRISK_SERVICE=1 -DENABLE_WEBM_VIDEO_CODECS=1 -DUSE_CBE=1 -DUSE_PMLOG=1 -DUSE_NEVA_SUSPEND_MEDIA_CAPTURE=1 -DUSE_FILESCHEME_CODECACHE=1 -DWEBOS_SUBMISSION_NUMBER=13 -DUSE_LOCAL_STORAGE_TRACKER=1 -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_OZONE=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DZLIB_IMPLEMENTATION -DADLER32_SIMD_NEON -DINFLATE_CHUNK_SIMD_NEON -DCRC32_ARMV8_CRC32 -DARMV8_OS_LINUX -I../../git/src/neva -Igen/neva -I../../git/src -Igen -I../../git/src/third_party/zlib -Wno-unused-variable -fno-ident -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -funwind-tables -fPIC -pipe -pthread -march=armv7-a -mfloat-abi=hard -mtune=generic-armv7-a -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -mfpu=neon -mthumb -O2 -fdata-sections -ffunction-sections -fno-omit-frame-pointer -g0 -fvisibility=hidden -Wno-psabi -Wno-unused-local-typedefs -Wno-maybe-uninitialized -Wno-deprecated-declarations -Wno-comments -Wno-packed-not-aligned -Wno-attributes -Wno-missing-field-initializers -Wno-unused-parameter -O3 -fdata-sections -ffunction-sections -std=gnu11 -c ../../git/src/third_party/zlib/gzclose.c -o obj/third_party/zlib/zlib/gzclose.o
cc1: warning: switch '-mcpu=cortex-a8' conflicts with switch '-march=armv7-a+simd'
In file included from /OE/build/luneos-mickledore/webos-ports/tmp-glibc/work/cortexa8t2hf-neon-webos-linux-gnueabi/webruntime/94-r59.2/recipe-sysroot/usr/include/features.h:392,
                 from /OE/build/luneos-mickledore/webos-ports/tmp-glibc/work/cortexa8t2hf-neon-webos-linux-gnueabi/webruntime/94-r59.2/recipe-sysroot/usr/include/bits/libc-header-start.h:33,
                 from /OE/build/luneos-mickledore/webos-ports/tmp-glibc/work/cortexa8t2hf-neon-webos-linux-gnueabi/webruntime/94-r59.2/recipe-sysroot/usr/include/stdio.h:27,
                 from ../../git/src/third_party/zlib/gzguts.h:21,
                 from ../../git/src/third_party/zlib/gzclose.c:6:
/OE/build/luneos-mickledore/webos-ports/tmp-glibc/work/cortexa8t2hf-neon-webos-linux-gnueabi/webruntime/94-r59.2/recipe-sysroot/usr/include/features-time64.h:26:5: error: #error "_TIME_BITS=64 is allowed only with _FILE_OFFSET_BITS=64"
   26 | #   error "_TIME_BITS=64 is allowed only with _FILE_OFFSET_BITS=64"
      |     ^~~~~

Upstream-Status: Submitted [https://github.com/madler/zlib/pull/764]

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/third_party/zlib/gzguts.h | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/third_party/zlib/gzguts.h b/src/third_party/zlib/gzguts.h
index 990a4d2514..a03d500507 100644
--- a/src/third_party/zlib/gzguts.h
+++ b/src/third_party/zlib/gzguts.h
@@ -3,15 +3,6 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 
-#ifdef _LARGEFILE64_SOURCE
-#  ifndef _LARGEFILE_SOURCE
-#    define _LARGEFILE_SOURCE 1
-#  endif
-#  ifdef _FILE_OFFSET_BITS
-#    undef _FILE_OFFSET_BITS
-#  endif
-#endif
-
 #ifdef HAVE_HIDDEN
 #  define ZLIB_INTERNAL __attribute__((visibility ("hidden")))
 #else
